pkg/systemd/quadlet/quadlet.go

```diff
@@ -1855,23 +1862,39 @@ func addNetworks(quadletUnitFile *parser.UnitFile, groupName string, serviceUnit
 	for _, network := range networks {
 		if len(network) > 0 {
 			quadletNetworkName, options, found := strings.Cut(network, ":")
-			if strings.HasSuffix(quadletNetworkName, ".network") {
-				// the podman network name is systemd-$name if none is specified by the user.
-				networkUnitInfo, ok := unitsInfoMap[quadletNetworkName]
+
+			isNetworkUnit := strings.HasSuffix(quadletNetworkName, ".network")
+			isContainerUnit := strings.HasSuffix(quadletNetworkName, ".container")
+
+			if isNetworkUnit || isContainerUnit {
+				// the podman network/container name is systemd-$name if none is specified by the user.
```

Comment:
I think I forgot to remove this comment, I don't think its relevant anymore because the value is expected to be set